lib/src/intl/date_format_field.dart

```diff
@@ -428,15 +428,39 @@ class _DateFormatPatternField extends _DateFormatField {
   ///
   /// This method handles reading any of the numeric fields. The [offset]
   /// argument allows us to compensate for zero-based versus one-based values.
-  void handleNumericField(IntlStream input, void Function(int) setter,
+  void handleNumericField(StringIterator input, void Function(int) setter,
       [int offset = 0]) {
-    var result = input.nextInteger(
-        digitMatcher: parent.digitMatcher,
-        zeroDigit: parent.localeZeroCodeUnit);
-    if (result == null) throwFormatException(input);
+    var result = nextInteger(
+      input,
+      parent.digitMatcher,
+      parent.localeZeroCodeUnit,
+    );
     setter(result + offset);
   }
 
+  /// Read as much content as [digitMatcher] matches from the current position,
```

Comment:
I removed this method from the `StringIterator` (formely known as `IntlStream`) class, as it is very specific, which runs against the very general nature of the `StringIterator`.